starter/go.mod

```diff
@@ -1,12 +0,0 @@
-module github.com/aerogear/charmil/starter
-
-go 1.16
-
-require (
-	github.com/aerogear/charmil v0.2.0
-	github.com/spf13/cobra v1.2.1
-	github.com/spf13/pflag v1.0.5
-	golang.org/x/text v0.3.6
-)
-
-replace github.com/aerogear/charmil => ../../charmil
```

Comment:
As the starter will be using some released version of the core, so we'll need it to be a sub-module. Which is why `go.mod` will be required here

Reply:
does it matter? as this will be most likely wiped and transformed into templates

Reply:
Yes, but there's still quite some time before we start working on templates, right?
Like we have to refine core and starter then figure out the templating engine to use etc. (Ref: https://github.com/aerogear/charmil/issues/164#issuecomment-883185917)

So maybe till then, we can leave this as a module.
Anyway, the issue of sub-module not working doesn't exist anymore after adding that `replace` directive on line 12, so I don't think keeping this for now will hurt us in any way. WDYT?

Please correct me if I'm wrong

Reply:
Also, we can include a `go.mod` file in the template too. WDYT?

Reply:
done

Reply:
that will make shell unusable. lets disscus

Reply:
> CLI is gimmick. starter will not be starter if it doest have all files needed to compile

Agreed. 

So I think we need to keep starter and templates as something separate
(Rules out the option 2 here: https://github.com/aerogear/charmil/issues/164#issuecomment-884060416)